core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -596,50 +598,63 @@ public static SqlOperandTypeChecker variadic(
       };
 
   /** Operand type-checking strategy that the first argument is of string type,
-   * and the remaining arguments can be of string or array of string type. */
-  public static final SqlOperandTypeChecker STRING_FIRST_STRING_ARRAY_OPTIONAL =
-      new SqlOperandTypeChecker() {
-        @Override public boolean checkOperandTypes(
-            SqlCallBinding binding,
+   * and the remaining arguments can be any type. */
+  public static final SqlOperandTypeChecker STRING_FIRST_OBJECT_REPEAT =
+      new StringFirstAndRepeatOperandTypeChecker(
+          SqlOperandCountRanges.from(2), "(<STRING>(,<ANY>)+)") {
+        @Override public boolean checkRepeatOperandTypes(SqlCallBinding callBinding,
             boolean throwOnFailure) {
-          // Check first operand is String type
-          if (!STRING.checkSingleOperandType(binding, binding.operand(0), 0, throwOnFailure)) {
-            return false;
+          ImmutableList.Builder<SqlTypeFamily> builder = ImmutableList.builder();
+          for (int i = 0; i < callBinding.getOperandCount(); i++) {
+            TypeCoercion coercion = callBinding.getValidator().getTypeCoercion();
+            RelDataType operandType = callBinding.getOperandType(i);
+            RelDataType cast =
+                ((AbstractTypeCoercion) coercion).implicitCast(operandType, SqlTypeFamily.STRING);
+            SqlTypeFamily family =
+                cast != null ? SqlTypeFamily.STRING
+                    : operandType.getSqlTypeName().getFamily();
+            assert family != null;
+            builder.add(family);
           }
+          ImmutableList<SqlTypeFamily> families = builder.build();
+          return family(families).checkOperandTypes(callBinding, throwOnFailure);
+        }
+      };
+
+  /** Operand type-checking strategy that the first argument is of string type,
+   * and the remaining arguments can be of string or array of string type. */
+  public static final SqlOperandTypeChecker STRING_FIRST_STRING_ARRAY_REPEAT =
```

Comment:
If it is unrelated, we can make another issue to do this.

Reply:
It is related, `StringFirstAndRepeatOperandTypeChecker` is refactored and defined to verify the `<String[, ANY]> `type, so that both spark and postgresql can use it.

Reply:
Ok. The code looks better